src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -215,22 +180,137 @@ private CommandResult executeCommand(String commandText) throws CommandException
         }
     }
 
-    private void executeUiCommand(CommandResult commandResult) {
+    private void executeUiCommand(CommandResult commandResult) throws CommandException {
         switch (commandResult.getUiCommand()) {
         case EXIT_APPLICATION:
             handleExit();
             break;
         case OPEN_HELP_WINDOW:
             handleHelp();
             break;
+        case DISPLAY_PROJECT:
+            handleDisplayProject(commandResult.getIndexOfProject());
+            break;
         case SHOW_CONTACTS:
             handleShowContactsTab();
             break;
         case SHOW_PROJECTS:
             handleShowProjectsTab();
             break;
+        case SHOW_EVENTS:
+            handleShowEventsTab();
+            break;
+        case SHOW_DEADLINES:
+            handleShowDeadlinesTab();
+            break;
+        case SHOW_TODOS:
+            handleShowTodosTab();
+            break;
+        case SHOW_PARTICIPANTS:
+            handleShowParticipantsTab();
+            break;
         default:
             assert false : "Command result should not be invalid";
         }
     }
+
+    // Handlers for UI Commands
+
+    /**
+     * Opens the help window or focuses on it if it's already opened.
+     */
+    @FXML
+    public void handleHelp() {
+        if (!helpWindow.isShowing()) {
+            helpWindow.show();
+        } else {
+            helpWindow.focus();
+        }
+    }
+
+    /**
+     * Closes the application.
+     */
+    @FXML
+    private void handleExit() {
+        GuiSettings guiSettings = new GuiSettings(primaryStage.getWidth(), primaryStage.getHeight(),
+                (int) primaryStage.getX(), (int) primaryStage.getY());
+        logic.setGuiSettings(guiSettings);
+        helpWindow.hide();
+        primaryStage.hide();
+    }
+
+    /**
+     * Displays the project at the current index.
+     * @param index Index of project to display.
+     */
+    private void handleDisplayProject(Index index) {
+        assert index != null;
+
+        if (projectDisplayPanel == null) {
+            projectDisplayPanel = new ProjectDisplayPanel();
+            infoDisplayPlaceholder.getChildren().add(projectDisplayPanel.getRoot());
+        }
+
+        projectDisplayPanel.displayProject(logic.getFilteredProjectsList().get(index.getZeroBased()), index);
+    }
+
+    /**
+     * Shows projects tab.
+     */
+    public void handleShowProjectsTab() {
+        tabPane.getSelectionModel().select(0);
```

Comment:
0 here sounds a little bit like a magic number here, should you put another variable for it?